lambdas/cfn-cognito-resource-server/index.js

```diff
@@ -0,0 +1,64 @@
+const AWS = require('aws-sdk');
+const notifyCFN = require('./notify-cfn')
+
+
+exports.handler = async (event, context) => {
+  try {
+    let cognitoIdentityServiceProvider = new AWS.CognitoIdentityServiceProvider(),
+      responseData,
+      physicalResourceId,
+      scopes = [{
+        ScopeDescription: 'Normal user access',
+        ScopeName: 'user.access'
+      }]
+
+    switch (event.RequestType) {
+      case 'Create':
+        if(event.ResourceProperties.Admin) {
+          scopes.push({
+            ScopeDescription: 'Admin access',
+            ScopeName: 'admin.access'
+          })
+        }
+
+        responseData = await cognitoIdentityServiceProvider.createResourceServer({
+          Identifier: event.ResourceProperties.Identifier,
+          UserPoolId: event.ResourceProperties.UserPoolId,
+          Name: event.ResourceProperties.Identifier,
+          Scopes: scopes
+        }).promise()
+
+        if (responseData && responseData.UserPoolId && responseData.Identifier) {
+          physicalResourceId = `${ responseData.UserPoolId }:${ responseData.Identifier }`
+        }
+
+        break;
+
+      case 'Update':
+        // the only param is userPoolId, and we don't expect/want customers to change that
+        // so just notify CFN that we succeeded
+
+        break;
+
+      case 'Delete':
+        console.log('This here is a delete!')
+        console.log('cognitoIdentityServiceProvider.deleteResourceServer:' + cognitoIdentityServiceProvider.deleteResourceServer)
+        responseData = await cognitoIdentityServiceProvider.deleteResourceServer({
+          Identifier: event.ResourceProperties.Identifier,
+          UserPoolId: event.ResourceProperties.UserPoolId
+        });
```

Comment:
should this also be promisified with `.promise()` ?

Reply:
Yes! Good catch!